portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -15,29 +15,95 @@
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+
+
+/** Servlet that returns some example content and handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  private List<String> messages; 
+  /** This class creates a storage for comments. 
+  *
+  */
+  private class CommentStore {
+  private String name;
+  private String message; 
+  private long timestamp;
+  
+  CommentStore(String name, String message, long timestamp){
+    this.name = name;
+    this.message = message;
+    this.timestamp = timestamp;
+    }
+  }
+  
   @Override
-  public void init() {
-    messages = new ArrayList<>();
-    messages.add("Hello"); 
-    messages.add("World"); 
-    messages.add("Love"); 
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException { 
+    List<CommentStore> commentData = new ArrayList<>();
+
+    // a query is used to extract data from the database 
+    Query query = new Query("comment").addSort("timestamp", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    for (Entity entity : results.asIterable()) {
+      String name = (String) entity.getProperty("name");
+      String message = (String) entity.getProperty("message");
+      long timestamp = (long) entity.getProperty("timestamp");
+
+      CommentStore comment = new CommentStore(name, message, timestamp);
+      commentData.add(comment);
+    }
+    // convert comments to json string
+    String json = convertToJsonUsingGson(commentData); 
+
+    // Send the JSON as the response 
+    response.setContentType("application/json;");
+    response.getWriter().println(json); 
+  }
+
+  private String convertToJsonUsingGson(List<CommentStore> commentData) {
+    Gson gson = new Gson();
+    String json = gson.toJson(commentData);
+    return json;
   }
 
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String message = messages.get((int) (Math.random() * messages.size())); 
-    response.setContentType("text/html;");
-    response.getWriter().println(message);
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String name = getParameter(request, "name-input", "");
+    String message = getParameter(request, "text-input", "");
+    long timestamp = System.currentTimeMillis();
+
+    Entity taskEntity = new Entity("comment");
+    taskEntity.setProperty("name", name);
+    taskEntity.setProperty("message", message);
+    taskEntity.setProperty("timestamp", timestamp);
+    
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```

Comment:
This appears several times. Does it make sense to make it a member of DataServlet?